CAS-1065 [Docs] Restructure ChannelsHeader component docs (#1992)
GetStream/stream-chat-android: docusaurus/docs/Android/03-ui/04-components/02-channel-list-header.md

# Channel List Header

> ## Overview

Reviewer: ```suggestion
```

> `ChannelListHeaderView` is a component designed to be displayed at the channels screen of your app. 

Reviewer 2: I see where I should put the View, but not what it does, which I fell that is missing here. Something like: "This is the view that shows the connection status, the title of the channel or conversation name..." would be nice.

Author: Explained this more precisely

> `ChannelListHeaderView` is a component designed to be displayed at the channels screen of your app. 

Reviewer: ```suggestion
`ChannelListHeaderView` is a component designed to be displayed on the channels screen of your app. 
```

It consists of the following elements: _User avatar_, _Title view_, _Action button_.
 
 | Light Mode | Dark Mode |
 | --- | --- |
 |![Light_mode](../../assets/channels_header.png)|![Dark_mode](../../assets/channels_header_dark.png)|
 
## Usage

To use `ChannelListHeaderView` in your layout, include it in your XML layout as shown below:
```XML
<io.getstream.chat.android.ui.channel.list.header.ChannelListHeaderView
    android:id="@+id/channelListHeaderView"
    android:layout_width="match_parent"
    android:layout_height="wrap_content"
    />
```

`ChannelListHeaderView` is supposed to work with `ChannelListHeaderViewModel`. 

> The basic setup of the view model and connecting it with the view can be done in the following way:

Reviewer: ```suggestion
The basic setup of the ViewModel and connecting it with the view can be done in the following way:
```

 ```kotlin
// Instantiate the view model 
val channelListHeaderViewModel: ChannelListHeaderViewModel by viewModels()
// Bind the view model with ChannelListView 
channelListHeaderViewModel.bindView(chanelHeaderView, viewLifecycleOwner)
// Optional: setup listeners for user avatar and action button 
chanelHeaderView.setOnActionButtonClickListener {
    navigateSafely(R.id.action_homeFragment_to_addChannelFragment)
}
chanelHeaderView.setOnUserAvatarClickListener {
    binding.drawerLayout.openDrawer(GravityCompat.START)
}
```
All the logic of subscribing to data emitted by view model is provided by the `ChannelListHeaderViewModel::bindView` function. 

> By default, view model will make view display avatar of the currently logged-in user and call the view to display "Searching for network" state when needed.

Reviewer: ```suggestion
By default, the ViewModel will make the view display avatar of the currently logged-in user and call the view to display "Searching for network" state when needed.
```

# Channel List Header

`ChannelListHeaderView` is a component that shows the title of the channels list, current connection status, avatar of the current user, 
and provides an action button which can be used to create a new conversation. 
It is designed to be displayed at the top of the channels screen of your app. 
It consists of the following elements: _User avatar_, _Title view_, _Action button_.
 
 | Light Mode | Dark Mode |
 | --- | --- |
 |![Light_mode](../../assets/channels_header.png)|![Dark_mode](../../assets/channels_header_dark.png)|
 
## Usage

To use `ChannelListHeaderView` in your layout, include it in your XML layout as shown below:
```XML
<io.getstream.chat.android.ui.channel.list.header.ChannelListHeaderView
    android:id="@+id/channelListHeaderView"
    android:layout_width="match_parent"
    android:layout_height="wrap_content"
    />
```

`ChannelListHeaderView` is supposed to work with `ChannelListHeaderViewModel`. 
The basic setup of the ViewModel and connecting it with the view can be done in the following way:
 ```kotlin

> // Instantiate the view model 

Reviewer 3: ```suggestion
// Instantiate the ViewModel
```

val channelListHeaderViewModel: ChannelListHeaderViewModel by viewModels()

> // Bind the view model with ChannelListView 

Reviewer 3: ```suggestion
// Bind the ViewModel with ChannelListView 
```

channelListHeaderViewModel.bindView(chanelHeaderView, viewLifecycleOwner)
// Optional: setup listeners for user avatar and action button 
chanelHeaderView.setOnActionButtonClickListener {

>     navigateSafely(R.id.action_homeFragment_to_addChannelFragment)

Reviewer 3: Hm, `navigateSafely` is something only relevant to our sample app. IMO, better just to leave a comment that you can navigate to create channel screen here.

Author: Fixed

}
chanelHeaderView.setOnUserAvatarClickListener {

>     binding.drawerLayout.openDrawer(GravityCompat.START)

Reviewer 3: Same, seems that nav drawer is something that is only relevant to our sample app

Author: Fixed

}
```

> All the logic of subscribing to data emitted by view model is provided by the `ChannelListHeaderViewModel::bindView` function. 

Reviewer 3: ```suggestion
All the logic of subscribing to data emitted by the ViewModel is provided by the `ChannelListHeaderViewModel::bindView` function. 
```

    // Handle User Avatar Click
}
```
If you want to hide those views you need to add XML attributes that modify their visibility. This is explained in the next section.

## Customizations

### Customization with XML Attributes

The appearance of `ChannelListHeaderView` can be conveniently modified using its XML attributes.   
```xml
    <io.getstream.chat.android.ui.channel.list.header.ChannelListHeaderView
        android:id="@+id/channelListHeaderView"
        android:layout_width="match_parent"
        android:layout_height="wrap_content"
        app:streamUiShowUserAvatar="false"
        app:streamUiShowOfflineProgressBar="false"
        app:streamUiActionButtonIcon="@drawable/ic_stream_logo"
        app:streamUiOnlineTitleTextStyle="bold"
        />
```

> The example above, hides the avatar view, makes the title text bold and sets the drawable of the action button to a drawable specified.

Reviewer: ```suggestion
The example above hides the avatar view, makes the title text bold and sets the drawable of the action button to a drawable specified.
```
